sopel/modules/help.py

```diff
@@ -1,350 +1,18 @@
-                msgfun(line)
-            if bot.doc[name][1]:
-                # Build a nice, grammatically-correct list of examples
-                examples = ', '.join(bot.doc[name][1][:-2] + [' or '.join(bot.doc[name][1][-2:])])
-                msgfun('e.g. ' + examples)
-    else:
-        # This'll probably catch most cases, without having to spend the time
-        # actually creating the list first. Maybe worth storing the link and a
-        # heuristic in the DB, too, so it persists across restarts. Would need a
-        # command to regenerate, too...
-        if (
-            'command-list' in bot.memory and
-            bot.memory['command-list'][0] == len(bot.command_groups) and
-            is_cache_valid(bot)
-        ):
-            url = bot.memory['command-list'][1]
-        else:
-            respond("Hang on, I'm creating a list.")
-            msgs = []
-
-            name_length = max(6, max(len(k) for k in bot.command_groups.keys()))
-            for category, cmds in collections.OrderedDict(sorted(bot.command_groups.items())).items():
-                category = category.upper().ljust(name_length)
-                cmds = set(cmds)  # remove duplicates
-                cmds = '  '.join(cmds)
-                msg = category + '  ' + cmds
-                indent = ' ' * (name_length + 2)
-                # Honestly not sure why this is a list here
-                msgs.append('\n'.join(textwrap.wrap(msg, subsequent_indent=indent)))
-
-            url = create_list(bot, '\n\n'.join(msgs))
-            if not url:
-                return
-            bot.memory['command-list'] = (len(bot.command_groups), url)
-            update_cache(bot)
-        respond("I've posted a list of my commands at {0} - You can see "
-                "more info about any of these commands by doing {1}help "
-                "<command> (e.g. {1}help time)"
-                .format(url, bot.config.core.help_prefix))
-
-
-def create_list(bot, msg):
-    """Creates & uploads the command list.
-
-    Returns the URL from the chosen pastebin provider.
-    """
-    msg = 'Command listing for {}{}\n\n{}'.format(
-        bot.nick,
-        ('@' + bot.config.core.host) if bot.config.help.show_server_host else '',
-        msg)
-
-    try:
-        result = PASTEBIN_PROVIDERS[bot.config.help.output](msg)
-    except PostingException:
-        bot.say("Sorry! Something went wrong.")
-        LOGGER.exception("Error posting commands")
-        return
-    return result
-
-
-@plugin.rule('$nick' r'(?i)help(?:[?!]+)?$')
-@plugin.priority('low')
-def help2(bot, trigger):
-    response = (
-        "Hi, I'm a bot. Say {1}commands to me in private for a list "
-        "of my commands, or see https://sopel.chat for more "
-        "general details. My owner is {0}."
-        .format(bot.config.core.owner, bot.config.core.help_prefix))
-    bot.reply(response)
+    LOGGER.warning(
+        'Sopel help built-in module is obsolete. '
```

Comment:
```suggestion
        'Sopel's built-in help plugin is obsolete. '
```

How many times have we talked about why "module" is confusing? 😝